backend/src/services/user.service.ts

```diff
@@ -51,4 +52,32 @@ export class UserService {
     public getAll(): Promise<User[]> {
         return User.findAll();
     }
+
+    public delete(req: Request, userId: number): Promise<number> {
+        if (!this.preconditionsDelete(parseInt(req.params.id, 10), userId)) {
+            return Promise.reject({message : 'You are not authorized'});
+        } else {
+            return User.destroy({
+                where : {userId : userId}
+            }).then(num => {
+                return Promise.resolve(num);
+            }).catch(err => Promise.reject({message: err}));
+        }
+    }
+    protected preconditionsDelete(deleterId: number, userToDeleteId: number): boolean {
+        return deleterId === userToDeleteId;
+    }
+}
+
+export class AdminService extends  UserService {
+    protected preconditionsDelete(deleterId: number, userToDeleteId: number): boolean {
+        User.findByPk(deleterId).then(usr => {
+            if (usr.isAdmin === 1) {
+                return true;
+            } else {
+                return super.preconditionsDelete(deleterId, userToDeleteId);
+            }
+        });
+        return super.preconditionsDelete(deleterId, userToDeleteId);
+    }
```

Comment:
Something here is wrong:

1. I got an error when a user wanted to delete himself ({"code":"ERR_HTTP_INVALID_STATUS_CODE"} and also error on the npm log)
2. If I am logged in as an Admin I am not able to delete other users

Reply:
Additionally to 1.: The user gets deleted though